kolibri/core/discovery/utils/network/client.py

```diff
@@ -64,6 +64,11 @@ def build_for_address(cls, address, timeout=None):
         :return: A NetworkClient with a verified connection
         :rtype: NetworkClient|cls
         """
+
+        # we cannot pass None address to get_normalized_url_variations()
+        # as it uses parse_address_into_components() which takes no None
+        if address is None:
```

Comment:
I found some places in code using requests that the baseurl is none, so to migrate to NetworkClient without raising any error, I did this.

Reply:
Are these the cases where the baseurl is the server itself? Maybe for that case we should make a `build_for_localhost` method or something to capture this case?

Reply:
By server you mean `https://studio.learningequality.org` ? 
One place i can recall where i observed the baseurl is None is [this](https://studio.learningequality.org), this happens when from the browser only we try to import channel from studio. 
So this behaviour(base being NONE) is already handled in `get_content_server_url` which our `_make_channel_endpoint_request` eventually calls to stack.

So for these cases instead of building network client from build_for_address which will throw an error is address is None. Can we simply create a NetworkClient obj as:
```
client = NetworkClient(baseurl)  # it deals with baseurl being null right?
```